### DISCUSSION TITLE
Two frigate instances

### DISCUSSION THREAD

**Discussion by marcus905** [blakeblackshear/frigate#9681] [Ask A Question] [answered]
Hi!

I was wondering about a particular usecase I have for frigate in my plans.

My hypothetical situation would be

- a "high performance" x86_64 router (10GBE WAN, 10/25GBE lan) that has 2 coral TPUs, and an Intel UHD 770 GPU for transcoding. Local storage would be a 2TB SSD.
- a home assistant yellow, to manage the rest of the IOT infra and to manage real time events from frigate
- a 4x12TB 2x25GBE NAS with an Intel UHD 630 GPU that would ultimately store the recordings
- a few cameras (12-16 right now) connected through a 48 port gigabit POE switch with some 10/25 ports and 40 GBE uplink, in a specific VLAN that's not connected to internet, and powered with a 6KVA double conversion UPS.

I'd ideally like to use the router to receive all camera input and process it locally (decoding/transcoding for the corals) as much as possible for near-real-time data (possibly using the 2TB SSD in there for first line storage), but store them persistently on the NAS so that it would execute the necessary transcoding to play/export the recordings when needed.

This would make the workload on the router static frigate-wise and leave the NAS do the extra work to avoid loading it.

Does this make sense? Am I going overboard with all this?

What would you do in this case?

**Comment by NickM-27** [comment, marked as answer]
there is no way to distribute frigate tasks across multiple devices and I don't really see a reason for this to be honest. 

The bulk of the work is in decoding the camera frames and running object detection. There is no transcoding done on recordings playback and the only transcoding done is when exporting a timelapse.

**Reply by marcus905** [reply]
Okay so this does away with the use of the NAS GPU then. I'd do everything in the router HW, correct?

**Reply by NickM-27** [reply]
based on the info provided, that is what I would recommend

**Reply by marcus905** [reply]
Thank you very much!

**Reply by luckylinux** [reply]
@NickM-27: I know this is somewhat of a old-ish Discussion, but I'd like to disagree on this :smiley:.

> there is no way to distribute frigate tasks across multiple devices and I don't really see a reason for this to be honest.

Unfortunately I couldn't get my Hands on a Ryzen PRO-series APU (for ECC Memory Support) so I got (another) AMD 5950x System (this Time on a B550 Motherboard) with an AMD RX 6600 GPU to do the Transcoding plus 2 x Hailo 8L to help with Object Detection (when Hailo 8L will be supported by Frigate, IIRC it was scheduled for 0.15). All on a Proxmox VE Series Hypervisor which is also handling some NAS-like Functions over NFS.

With that setup, the use case for distributed/separated Frigate Services is:
- Transcoding: `amdgpu` Driver crashes Proxmox VE Kernel as soon as boot starts -> Need to blacklist `amdgpu` Driver and let it run in a VM
- Object Detection: The 2 x Hailo 8L are bound to the B550 Chipset (NOT the CPU PCIe Lanes), meaning that they CANNOT be passed-through a VM, so I would need to use LXC Container for it (or Podman Directly on the Host)

So, as it currently is, it doesn't appear like I can run both "Services" on the same Frigate Instance :cry:. And I don't think/know if it's simply possible to let 2 instances of Frigate communicate to each other when there is a need to e.g. save a Video Stream.

Basically similarly to what Doubletake does/did communicating over MQQT :smiley:.

**Comment by cliff-rosenberg** [comment]
> Am I going overboard with all this?

Maybe so. But that's the nice thing about "DIY", you can do whatever your budget allows and make things conform to what YOU like. I do large scale manufacturing automation for my "day job", so my home projects often go further into the higher performance options because that's what I use daily at work. The hard thing is keeping the "Wife Acceptance Factor" in the green range. LOL
